ipympl/backend_nbagg.py

```diff
@@ -198,12 +198,6 @@ class Canvas(DOMWidget, FigureCanvasWebAggCore):
     _force_full = Bool()
     _current_image_mode = Unicode()
     _dpi_ratio = Float(1.0)
-    _is_idle_drawing = Bool()
-    _is_saving = Bool()
-    _button = Any()
-    _key = Any()
-    _lastx = Any()
-    _lasty = Any()
```

Comment:
Just wondering if those are use in core Matplotlib, do you know @tacaswell?

Reply:
Yes, we use them for preventing re-entrant draws (`_is_idle_drawing` and `_is_saving`) and assembling the composite events that eventually propagates through our event system (because the keyboard and mouse events come in separate and we merge them).

Reply:
So is it safe removing them?

Reply:
The canvas still has all of these attributes that it inherits from webagg. So this runs without throwing an error:
 ```python
for attr in ['_lastx', '_lasty', '_button', '_key', '_is_saving', '_is_idle_drawing']:
    assert hasattr(fig.canvas, attr)
```
so I think it is safe to remove them.